tests/conftest.py

```diff
@@ -2043,9 +2025,116 @@ def saved_hyperopt_results():
             'is_best': False
             }
     ]
-
     for res in hyperopt_res:
         res['results_metrics']['holding_avg_s'] = res['results_metrics']['holding_avg'
                                                                          ].total_seconds()
-
     return hyperopt_res
+
+# * Margin Tests
+# TODO-mg: fill in these tests with something useful
+
+
+@pytest.fixture
+def ten_minutes_ago():
+    return datetime.utcnow() - timedelta(hours=0, minutes=10)
+
+
+@pytest.fixture
+def five_hours_ago():
+    return datetime.utcnow() - timedelta(hours=1, minutes=0)
+
+
+@pytest.fixture(scope='function')
+def limit_short_order_open():
+    return {
+        'id': 'mocked_limit_short',
+        'type': 'limit',
+        'side': 'sell',
+        'symbol': 'mocked',
+        'datetime': arrow.utcnow().isoformat(),
+        'timestamp': arrow.utcnow().int_timestamp,
+        'price': 0.00001173,
+        'amount': 90.99181073,
+        'borrowed': 90.99181073,
+        'filled': 0.0,
+        'cost': 0.00106733393,
+        'remaining': 90.99181073,
+        'status': 'open',
+        'is_short': True
+    }
+
+
+@pytest.fixture
+def limit_exit_short_order_open():
+    return {
+        'id': 'mocked_limit_exit_short',
+        'type': 'limit',
+        'side': 'buy',
+        'pair': 'mocked',
+        'datetime': arrow.utcnow().isoformat(),
+        'timestamp': arrow.utcnow().int_timestamp,
+        'price': 0.00001099,
+        'amount': 90.99181073,
+        'filled': 0.0,
+        'remaining': 90.99181073,
+        'status': 'open'
+    }
+
+
+@pytest.fixture(scope='function')
+def limit_short_order(limit_short_order_open):
+    order = deepcopy(limit_short_order_open)
+    order['status'] = 'closed'
+    order['filled'] = order['amount']
+    order['remaining'] = 0.0
+    return order
+
+
+@pytest.fixture
+def limit_exit_short_order(limit_exit_short_order_open):
+    order = deepcopy(limit_exit_short_order_open)
+    order['remaining'] = 0.0
+    order['filled'] = order['amount']
+    order['status'] = 'closed'
+    return order
+
+
+@pytest.fixture(scope='function')
+def market_short_order():
+    return {
+        'id': 'mocked_market_buy',
+        'type': 'market',
+        'side': 'buy',
+        'symbol': 'mocked',
+        'datetime': arrow.utcnow().isoformat(),
+        'price': 0.00004173,
+        'amount': 91.99181073,
+        'filled': 91.99181073,
+        'remaining': 0.0,
+        'status': 'closed',
+        'is_short': True,
+        'leverage': 3
```

Comment:
Are these 2 fields actually part of the Return values from ccxt? If not, this will create a false sense of "working" - while in reality we'll never get this value...

i'm asking this because i know they are not ... 
https://github.com/ccxt/ccxt/blob/39a5cd3d320fb8ea2935fa0061b668c1a0a0cc13/js/binance.js#L1983-L2005

Reply:
Do you have an example of some code I can execute to receive a valid ccxt response object?

Reply:
I know that with Kraken the leverage is passed in like 

```
kraken.create_market_sell_order('BTC/USD', 1, {'leverage': 2})
```

I'm not sure how that would transfer over to out Order Class

Reply:
the problem is - the code you're pointing at here is the INPUT to create a order.

if you run this code, you'll notice that 
a) you've gone short on 30.000+$ (so please be careful) 
b) the response - assuming the order is places successfully - will not contain leverage

at least not in the top level response (which is guaranteed by ccxt to be aligned between exchanges).
While you can reach into the info substructure - that will be different by exchange - so that should not be done from the trade class.

i'll need to think about this some - as i'm currently not clear how the (real world) responses from the exchanges differ - to determine the best course of action.

Technically - we should however keep all exchange-specifics in the exchange class(es) - otherwise it'll be scattered around the code which will be a nightmare to maintain.

so worst case - we call a dummy method which does nothing by default (from the exchange class) - and "extracts" the necessary keys (leverage, ...) from the info key and injects them into the order responses.

Reply:
Would it be a mistake to just specify the `leverage` and `is_short` when creating the trade object, and remove all the logic of assigning them from within the update function?

Reply:
no - i think that would be correct - that was my assumption anyway - as using leverage / borrowing is always a explicit decision at the time of trade entry.